client/blocks/edit-gravatar/index.jsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import { getCurrentUser } from 'state/current-user/selectors';
+import {
+	isCurrentUserUploadingGravatar,
+} from 'state/current-user/gravatar-status/selectors';
+import Gravatar from 'components/gravatar';
+import { isOffline } from 'state/application/selectors';
+import { localize } from 'i18n-calypso';
+import Spinner from 'components/spinner';
+import { getToken as getOauthToken } from 'lib/oauth-token';
+import { uploadGravatar } from 'state/current-user/gravatar-status/actions';
+
+/**
+ * Module dependencies
+ */
+const debug = debugFactory( 'calypso:edit-gravatar' );
+
+export class EditGravatar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.handleOnPick = this.handleOnPick.bind( this );
+	}
+
+	static propTypes = {
+		isOffline: PropTypes.bool,
+		isUploading: PropTypes.bool,
+		translate: PropTypes.func,
+		uploadGravatar: PropTypes.func,
+		user: PropTypes.object,
+	};
+
+	handleOnPick( files ) {
+		const { uploadGravatar: uploadGravatarAction, user } = this.props;
```

Comment:
my personal preference is to split this into multiple lines. normally with just two items I may not, but here we are renaming one and I could see enough reason to justify the clarification…

```js
const {
	uploadGravatar: uploadGravatarAction,
	user,
} = this.props;
```

Also, are we simply renaming this to prevent the `no-shadow` warning?